internal/tuf/tuf.go

```diff
@@ -206,10 +208,12 @@ func (d *Delegation) Matches(target string) bool {
 // Delegation defines the schema for a single delegation entry. It differs from
 // the standard TUF schema by allowing a `custom` field to record details
 // pertaining to the delegation.
+// Updated to comply with TUF's TAP 3
```

Comment:
For backwards compat, I suggest making this a separate struct. I want to introduce a `gittufVersion` field in metadata that we'll use to determine what the metadata looks like. Some repetition is fine for that.

Reply:
Or actually, we introduce a _separate_ struct for "legacy" (pre-teams) metadata so we can continue verifying them. And when a legacy metadata is loaded / manipulated for whatever reason, can we automatically convert it into this new schema so that over time we get even legacy metadata moved over? We obviously can't ever drop the legacy metadata struct/schema for verification from the start scenarios, but at least it'll simplify the logic of enabling teams when it's not actually used and so on.

Reply:
Let's make it its own PR. ATM, I think we could avoid tying it to a specific gittuf version and just have incrementing integers for the metadata schema, but maybe that introduces more complications. Can you sketch something out?